interactors/nonceHandlerV3/nonceTransactionsHandler.go

```diff
@@ -0,0 +1,195 @@
+		if err != nil {
+			return err
+		}
+
+		err = anh.ApplyNonceAndGasPrice(ctx, transactions...)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (nth *nonceTransactionsHandlerV3) getOrCreateAddressNonceHandler(address core.AddressHandler) (interactors.AddressNonceHandlerV3, error) {
+	anh := nth.getAddressNonceHandler(address)
+	if !check.IfNil(anh) {
+		return anh, nil
+	}
+
+	return nth.createAddressNonceHandler(address)
+}
+
+func (nth *nonceTransactionsHandlerV3) getAddressNonceHandler(address core.AddressHandler) interactors.AddressNonceHandlerV3 {
+	nth.mutHandlers.RLock()
+	defer nth.mutHandlers.RUnlock()
+
+	anh, found := nth.handlers[string(address.AddressBytes())]
+	if found {
+		return anh
+	}
+	return nil
+}
+
+func (nth *nonceTransactionsHandlerV3) createAddressNonceHandler(address core.AddressHandler) (interactors.AddressNonceHandlerV3, error) {
+	nth.mutHandlers.Lock()
+	defer nth.mutHandlers.Unlock()
+
+	addressAsString := string(address.AddressBytes())
+	anh, found := nth.handlers[addressAsString]
+	if found {
+		return anh, nil
+	}
+
+	anh, err := NewAddressNonceHandlerV3(nth.proxy, address, nth.pollingInterval)
+	if err != nil {
+		return nil, err
+	}
+	nth.handlers[addressAsString] = anh
+
+	return anh, nil
+}
+
+func (nth *nonceTransactionsHandlerV3) filterTransactionsBySenderAddress(transactions []*transaction.FrontendTransaction) map[string][]*transaction.FrontendTransaction {
+	filterMap := make(map[string][]*transaction.FrontendTransaction)
+	for _, tx := range transactions {
+		transactionsPerAddress, ok := filterMap[tx.Sender]
+		if !ok {
+			transactionsPerAddress = make([]*transaction.FrontendTransaction, 0)
+		}
+		filterMap[tx.Sender] = append(transactionsPerAddress, tx)
+	}
+
+	return filterMap
+}
+
+// SendTransactions will store and send the provided transaction
+func (nth *nonceTransactionsHandlerV3) SendTransactions(ctx context.Context, txs ...*transaction.FrontendTransaction) ([]string, error) {
+	g, ctx := errgroup.WithContext(ctx)
+	sentHashes := make([]string, len(txs))
+	for i, tx := range txs {
+		if tx == nil {
+			return nil, interactors.ErrNilTransaction
+		}
+
+		// Work with a full copy of the provided transaction so the provided one can change without affecting this component.
+		// Abnormal and unpredictable behaviors due to the resending mechanism are prevented this way
+		txCopy := *tx
+
+		addrAsBech32 := txCopy.Sender
+		address, err := data.NewAddressFromBech32String(addrAsBech32)
+		if err != nil {
+			return nil, fmt.Errorf("%w while creating address handler for string %s", err, addrAsBech32)
+		}
+
+		anh, err := nth.getOrCreateAddressNonceHandler(address)
+		if err != nil {
+			return nil, err
+		}
+
+		idx := i
+		g.Go(func() error {
+			sentHash, errSend := anh.SendTransaction(ctx, &txCopy)
+			if errSend != nil {
+				return fmt.Errorf("%w while sending transaction for address %s", errSend, addrAsBech32)
+			}
+
+			sentHashes[idx] = sentHash
```

Comment:
I believe you can write concurrently on a slice as long as you are providing an index, since you are not writing to the same variable, nor changing the slice header inside the `errgroup`. Further reads on the slice cannot be made until all goroutines are finished, hence the `g.Wait()` at the end. I've also tested this function with the -race condition and so far it hasn't returned any issues.